ext/standard/array.c

```diff
@@ -1889,6 +1884,15 @@ PHP_FUNCTION(extract)
 
 		if (Z_TYPE(final_name) == IS_STRING && php_valid_var_name(Z_STRVAL(final_name), Z_STRLEN(final_name))) {
 			zval *orig_var;
+
+			if (Z_STRLEN(final_name) == sizeof("this")-1  && !strcmp(Z_STRVAL(final_name), "this")) {
```

Comment:
it's better to use zend_string_equals(Z_STR(final_name), CG(known_string)[THIS_STR[)